### ISSUE TITLE
Ruby JSON output incorrectly formats int64/uint64/fixed64 as number

### ISSUE THREAD

**Issue by lstoll** [protocolbuffers/protobuf#6147]
**What version of protobuf and what language are you using?**
Version:3.7.1/master
Language: Ruby

**What operating system (Linux, Windows, ...) and version?**

OS X 10.14.5/Various Linxes

**What runtime / compiler are you using (e.g., python version or gcc version)**

Ruby 2.6.0, Ruby 2.4.5

**What did you do?**

Called `to_json` on a message containing int64 fields

**What did you expect to see**

Expected to see int64/uint64/fixed64 serialized as decimal strings, as per spec: https://developers.google.com/protocol-buffers/docs/proto3#json

**What did you see instead?**

Output was serialized as number

**Anything else we should know about your project / environment**

Earlier report for the deserialization: https://github.com/protocolbuffers/protobuf/issues/3013
PHP seems to use same serializer, open issue: https://github.com/protocolbuffers/protobuf/issues/5663

**Test case:**

#### `Gemfile`

```
source "https://rubygems.org"

gem "google-protobuf", "3.7.1"
gem "minitest"
```

#### `test.rb`

```
require 'minitest/autorun'
require 'google/protobuf'
require 'json'

class ProtoJSONTest < Minitest::Test
  pool = Google::Protobuf::DescriptorPool.new
  pool.build do
    add_message "JSONTester" do
      optional :int64, :int64, 1
      optional :uint64, :uint64, 2
      optional :fixed64, :fixed64, 3
    end
  end

  JSONTester = pool.lookup("JSONTester").msgclass

  def test_int64_as_string
    jt = JSONTester.new(int64: 1234567890, uint64: 1234567890, fixed64: 1234567890)
    jm = JSON.parse(jt.to_json)

    refute_nil jm["int64"]
    assert_kind_of String, jm["int64"]

    refute_nil jm["uint64"]
    assert_kind_of String, jm["uint64"]

    refute_nil jm["fixed64"]
    assert_kind_of String, jm["fixed64"]
  end
end
```

#### Output:

```
$ bundle exec ruby test.rb 
Run options: --seed 34900

# Running:

F

Finished in 0.001602s, 624.2197 runs/s, 1248.4394 assertions/s.

  1) Failure:
ProtoJSONTest#test_int64_as_string [test.rb:22]:
Expected 1234567890 to be a kind of String, not Integer.

1 runs, 2 assertions, 1 failures, 0 errors, 0 skips
```

**Comment by lstoll** [comment]
I believe the fix is going to be updating https://github.com/protocolbuffers/upb/blob/cf35baa/upb/json/printer.c#L211-L221 to split out the 32 from 64 bit usage, then making that output a string. I'll try and get a PR up for that.

**Comment by BSBandme** [comment]
Thanks for submitting fixing PR!

**Comment by TeBoring** [comment]
@lstoll have you finished the PR?

**Comment by TeBoring** [comment]
Fixed in #6251
